File: app/src/main/java/com/divercity/app/features/profile/selectindustry/industry/IndustryPaginatedRepository.java
package com.divercity.app.features.profile.selectindustry.industry;

import android.support.annotation.Nullable;

import com.divercity.app.core.utils.Listing;
import com.divercity.app.data.entity.industry.IndustryResponse;

/**
 * Created by lucas on 01/10/2018.
 */

public interface IndustryPaginatedRepository {

    Listing<IndustryResponse> fetchCompanies(@Nullable String query);

    void retry();

    void refresh();

    void clear();

}

File: app/src/main/java/com/divercity/android/features/industry/base/IndustryPaginatedRepository.kt
package com.divercity.android.features.industry.base

import com.divercity.android.core.base.datasource.BaseDataSourceRepository
import com.divercity.android.core.base.usecase.Params
import com.divercity.android.core.base.usecase.UseCase
import com.divercity.android.data.entity.industry.IndustryResponse
import com.divercity.android.features.industry.base.usecase.FetchIndustriesUseCase
import javax.inject.Inject

/**
 * Created by lucas on 01/10/2018.
 */

class IndustryPaginatedRepository @Inject
internal constructor(private val fetchIndustriesUseCase: FetchIndustriesUseCase) :
    BaseDataSourceRepository<IndustryResponse>() {

    override fun getUseCase(): UseCase<List<IndustryResponse>, Params> = fetchIndustriesUseCase
}
